agents/packages/ethlisbon/skills/price_prophet/behaviours.py

```diff
@@ -233,43 +229,23 @@ def async_act(self) -> Generator:
         with self.context.benchmark_tool.measure(self.behaviour_id).local():
             random_state = self.synchronized_data.period_count
             most_voted: JSONLike = self.get_strict(AnnotateDataRound.selection_key)
-            results_grid = self.train_model(pd.read_json(most_voted), random_state)
-            content, sender = results_grid.to_json(), self.context.agent_address
-            payload = TrainModelPayload(sender=sender, content=content)
+            try:
+                results_grid = train_model(pd.read_json(most_voted), random_state)
+                self.context.shared_state[TrainModelRound.selection_key] = results_grid
+            except Exception as e:
+                results_grid = None
+                self.context.logger.error(f"Failed to complete training forecaster: {e}")
+
+            sender = self.context.agent_address
+            payload = TrainModelPayload(sender=sender, content=bool(results_grid))
+            self.context.logger.info(f"Model trained: {bool(results_grid)}")
```

Comment:
we only return whether we successfully trained the model. It is stored in the `shared_state`